packages/backend-common/src/reading/AzureUrlReader.ts

```diff
@@ -19,22 +19,53 @@ import {
   readAzureIntegrationConfigs,
 } from '@backstage/integration';
 import fetch from 'cross-fetch';
+import { Readable } from 'stream';
+import parseGitUri from 'git-url-parse';
 import { NotFoundError } from '../errors';
-import { ReaderFactory, ReadTreeResponse, UrlReader } from './types';
+import {
+  ReaderFactory,
+  ReadTreeOptions,
+  ReadTreeResponse,
+  UrlReader,
+} from './types';
+import { ReadTreeResponseFactory } from './tree';
+
+export function getDownloadUrl(url: string): URL {
+  const {
+    name: repoName,
+    owner: project,
+    organization,
+    protocol,
+    resource,
+    filepath,
+  } = parseGitUri(url);
+
+  // scopePath will limit the downloaded content
+  // /docs will only download the docs folder and everything below it
+  // /docs/index.md will only download index.md but put it in the root of the archive
+  const scopePath = filepath ? `&scopePath=${filepath}` : '';
```

Comment:
Wish GitHub had this feature :grin:, you wanna `encodeURIComponent` here though

Reply:
Will fix. I thought that `URL` handled this in the end.